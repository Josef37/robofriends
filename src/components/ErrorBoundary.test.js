import React from "react";
import { shallow } from "enzyme";
import ErrorBoundary from "./ErrorBoundary";

describe("ErrorBoundary component", () => {
  it("renders nothing without error", () => {
    const wrapper = shallow(<ErrorBoundary />);
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it("renders error message when error occurs", () => {
    const wrapper = shallow(<ErrorBoundary />);
    wrapper.instance().componentDidCatch();
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
