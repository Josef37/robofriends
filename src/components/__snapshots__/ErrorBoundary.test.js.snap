// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary component renders error message when error occurs 1`] = `
"<h1>
  Ooooops! Something went wrong...
</h1>"
`;

exports[`ErrorBoundary component renders nothing without error 1`] = `""`;
